File: shoe_wash_app/mobile_app/lib/models/expense.dart
class Expense {
  final int id;
  final String date;
  final String category;
  final double amount;
  final String? note;

  Expense({
    required this.id,
    required this.date,
    required this.category,
    required this.amount,
    this.note,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'],
      date: json['date'],
      category: json['category'],
      amount: (json['amount'] as num).toDouble(),
      note: json['note'],
    );
  }
}

File: shoe_wash_app/backend/src/domain/expense.py
from dataclasses import dataclass
from datetime import date
from typing import Optional


class ExpenseCategory:
    """Plain string constants -- easy to add a new category without touching
    any other file."""

    SUPPLIES = "supplies"
    UTILITIES = "utilities"
    MACHINE_UPKEEP = "machine_upkeep"
    OTHER = "other"


@dataclass
class Expense:
    """One outgoing cost -- supplies, utilities, machine upkeep, or other."""

    id: Optional[int]
    date: date
    category: str
    amount: float
    note: Optional[str] = None

    def __post_init__(self):
        if self.amount <= 0:
            raise ValueError("Expense amount must be positive")
